### DISCUSSION TITLE
Mapping stack frame data to function variables

### DISCUSSION THREAD

**Discussion by mixern6** [NationalSecurityAgency/ghidra#6169] [Q&A] [answered]
I have a crash report generated by breakpad:

	 9  server.so + 0x00fea69f
	    rbp = 0x00007ffd3dc42c90   rsp = 0x00007ffd3dc42c60   rip = 0x00007f4dc53cd69f
	 
		Found by: previous frame's frame pointer
		Stack contents:	
		 000000003dc42c60	00 08 02 61 0c 7f 00 00 12 00 00 00 00 50 c3 47	...a.........P.G
		 000000003dc42c70	61 04 00 00 00 00 00 00 00 26 3f 20 0c 7f 00 00	a........&? ....
		 000000003dc42c80	ff ff ff ff 00 00 00 00 00 26 3f 20 0c 7f 00 00	.........&? ....
		 000000003dc42c90	c0 2c c4 3d fd 7f 00 00 3a d8 3c c5 4d 7f 00 00	.,.=....:.<.M...
	
	10  server.so + 0x00fea83a
	    rbp = 0x00007ffd3dc42cc0   rsp = 0x00007ffd3dc42ca0   rip = 0x00007f4dc53cd83a
	 
		Found by: previous frame's frame pointer
		Stack contents:	
		 000000003dc42ca0	00 26 3f 20 0c 7f 00 00 61 04 00 00 00 00 00 00	.&? ....a.......
		 000000003dc42cb0	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00	................
		 000000003dc42cc0	10 2d c4 3d fd 7f 00 00 58 41 be c4 4d 7f 00 00	.-.=....XA..M...
	
	11  server.so + 0x00801158
	    rbp = 0x00007ffd3dc42d10   rsp = 0x00007ffd3dc42cd0   rip = 0x00007f4dc4be4158
	 
		Found by: previous frame's frame pointer
		Stack contents:	
		 000000003dc42cd0	61 04 00 00 00 00 00 00 00 00 00 00 00 00 00 00	a...............
		 000000003dc42ce0	01 c7 03 00 00 00 00 00 00 26 3f 20 0c 7f 00 00	.........&? ....
		 000000003dc42cf0	61 04 00 00 00 00 00 00 00 00 00 00 00 00 00 00	a...............
		 000000003dc42d00	01 c7 03 00 00 00 00 00 ff ff ff ff 00 00 00 00	................
		 000000003dc42d10	a0 2e c4 3d fd 7f 00 00 fd 38 30 c5 4d 7f 00 00	...=.....80.M...

And the function from headless ghidra:

	// Call address: 0x00fea83a
	// Call line: ~7
	// Parameter count: 1
	// Parameter: [long * param_1@RDI:8]
	// Stack frame size: 36
	// Stack frame contents: 
	// >                  -28	local_1c (4 bytes)
	// >                  -16	local_10 (8 bytes)
	// > return offset    0	
	// > param offset     8	
	
	void FUN_010ea820(long *param_1)	
	{
	  int iVar1;
	  float fVar2;
	  float fVar3;
	  
	  iVar1 = FUN_010ea5b0(0.5,param_1);
	  if (0 < iVar1) {
	    fVar3 = 0.0;
	    if (*(char *)(param_1 + 0x177) == '\0') {
	      fVar3 = 0.5;
	    }
	    fVar2 = (float)FUN_00c1c550(*(undefined4 *)(param_1[2] + 0x38));
	    FUN_00900b50(fVar2 + fVar3,param_1,0);
	    return;
	  }
	  FUN_009008a0(0,param_1,0,0,0,0);
	  return;
	}

Is it possible to identify the values in **param_1**, **iVar1**, **fVar2** and **fVar3** ?
How are the **local_1c** and **local_10** related to the function variables? Why are they different in size and count?

**Comment by ghidra1** [comment, marked as answer]
The listing establishes local variables (i.e., stack analyzer) when it detects stack references, however these may not be true variables as determined by the decompiler (e.g., a single variable can jump between varnodes - comes in on register and moves to stack).  For this reason I personally like to disable stack analysis.
